openmetadata-service/src/main/java/org/openmetadata/service/resources/query/QueryResource.java

```diff
@@ -0,0 +1,520 @@
+            description = "Get List of queries",
+            content =
+                @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = QueryResource.QueryList.class)))
+      })
+  public ResultList<Query> listQueries(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(
+              description = "Fields requested in the returned resource",
+              schema = @Schema(type = "string", example = FIELDS))
+          @QueryParam("fields")
+          String fieldsParam,
+      @Parameter(description = "Limit the number queries returned. " + "(1 to 1000000, default = 10)")
+          @DefaultValue("10")
+          @Min(0)
+          @Max(1000000)
+          @QueryParam("limit")
+          int limitParam,
+      @Parameter(description = "Returns list of queries before this cursor", schema = @Schema(type = "string"))
+          @QueryParam("before")
+          String before,
+      @Parameter(description = "Returns list of queries after this cursor", schema = @Schema(type = "string"))
+          @QueryParam("after")
+          String after,
+      @Parameter(
+              description = "Include all, deleted, or non-deleted entities.",
+              schema = @Schema(implementation = Include.class))
+          @QueryParam("include")
+          @DefaultValue("non-deleted")
+          Include include)
+      throws IOException {
+    ListFilter filter = new ListFilter(include);
+    return super.listInternal(uriInfo, securityContext, fieldsParam, filter, limitParam, before, after);
+  }
+
+  @GET
+  @Path("/{id}")
+  @Operation(
+      operationId = "getQueryByID",
+      summary = "Get a query",
+      tags = "query",
+      description = "Get a Query by `id`",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "query",
+            content = @Content(mediaType = "application/json", schema = @Schema(implementation = Query.class))),
+        @ApiResponse(responseCode = "404", description = "Query for instance {id} is not found")
+      })
+  public Query getQueryById(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(description = "query Id", schema = @Schema(type = "UUID")) @PathParam("id") UUID id,
+      @Parameter(
+              description = "Fields requested in the returned resource",
+              schema = @Schema(type = "string", example = FIELDS))
+          @QueryParam("fields")
+          String fieldsParam,
+      @Parameter(
+              description = "Include all, deleted, or non-deleted entities.",
+              schema = @Schema(implementation = Include.class))
+          @QueryParam("include")
+          @DefaultValue("non-deleted")
+          Include include)
+      throws IOException {
+    return getInternal(uriInfo, securityContext, id, fieldsParam, include);
+  }
+
+  @GET
+  @Path("/name/{fqn}")
```

Comment:
What is fqn for a query? We should delete get-by-name methods if it is not applicable.

Reply:
fqn is evaluated as the name -> when created from UI , or the query checksum if the name is not provided